assembl/static2/js/app/components/synthesis/sideMenu.jsx

```diff
@@ -11,14 +11,16 @@ type SideMenuProps = {
   rootIdeas: Array<SynthesisIdea>,
   descendants: Array<SynthesisIdea>,
   synthesisPostId: string,
-  ideaOnScroll?: string
+  ideaOnScroll?: string,
+  show: boolean,
+  innerRef?: Function
 };
 
 const SideMenu = (props: SideMenuProps) => {
-  const { rootIdeas, descendants, synthesisPostId, ideaOnScroll } = props;
+  const { rootIdeas, descendants, synthesisPostId, ideaOnScroll, innerRef, show } = props;
   const slug = getDiscussionSlug();
   return (
-    <div className="synthesis-side-menu">
+    <div className="synthesis-side-menu" ref={innerRef} style={{ display: show ? 'block' : 'none' }}>
```

Comment:
Is there a reason to have it hidden by style instead of not adding it to the DOM ?

Reply:
Isn't it better in terms of performance ? At first I did this because I thought I could still measure the element's height if it's hidden but it's not possible. I just kept it like this

Reply:
about performance, I imagine it's the contrary because you don't render this part of the DOM. But maybe I am missing something @vincentfretin ?

Reply:
Destroying and remounting a component tree is far more expensive than just flipping a css property.  So what @MilanBarande did is ok.